.github/workflows/ci.yaml

```diff
@@ -0,0 +1,115 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+
+jobs:
+  cargo-test:
+    env:
+      IRONCORE_ENV: dev
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-18.04, macos-10.15, windows-2019]
+        features: [--features tls-vendored-openssl --no-default-features]
+        toolchain: [stable]
+        include:
+          # different features combinations
+          - os: ubuntu-18.04
+            features: --features beta
+          - os: ubuntu-18.04
+            features: --features blocking
+          - os: ubuntu-18.04
+            features: --features "blocking beta"
+          # different openssl implementations
+          - os: ubuntu-18.04
+            features: --features tls-rustls --no-default-features
+          - os: ubuntu-18.04
+            features: --features tls-default --no-default-features
+          # MSRV
+          - os: ubuntu-18.04
+            toolchain: 1.39.0
+    steps:
+      - uses: actions/checkout@v2
+      - if: matrix.toolchain != ''
```

Comment:
These machines actually already have Rust on them. So the only time I _really_ need to call the `actions-rs/toolchain` action is when I need something different than what they have (like if I need `cargofmt` or the `beta` toolchain). So this conditional makes it so that I don't have to add `toolchain: stable` to every different "include" build as it already has stable installed. 

As a side note, the reason that line 19 has `toolchain: [stable]` instead of skipping it is so that lines 34 and 35 _add_ to the build matrix instead of just modifying the ubuntu build. The `-include` is overloaded to do both, so it can lead to some confusing builds.